core/src/test/java/org/bitcoinj/core/TransactionTest.java

```diff
@@ -353,6 +353,18 @@ public void testWitnessSignatureP2SH_P2WPKH() {
         assertEquals(signedTxHex, HEX.encode(tx.bitcoinSerialize()));
         assertEquals(signedTxHex.length() / 2, tx.getMessageSize());
     }
+    
+    @Test
+    public void testWitnessSignatureP2SH_P2WPKH_Single() throws SignatureDecodeException {
```

Comment:
It looks like you took the example from https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#p2sh-p2wsh

If I am right, can you correct the title of this test and maybe add the URL in a comment like in the tests above.

Ideally we would implement the entire test vector(s), but it's not required for merging this.

Reply:
The hex could be separated a bit, like in the previous tests. But don't worry, I can do that too.